apollo-router/src/plugins/connectors/configuration.rs

```diff
@@ -25,6 +25,10 @@ pub(crate) struct ConnectorsConfig {
     /// The maximum number of requests for a connector source
     #[serde(default)]
     pub(crate) max_requests_per_operation_per_source: Option<usize>,
+
+    /// Configuration for HTTP response snapshots
+    #[serde(default)]
+    pub(crate) snapshot: SnapshotConfiguration,
```

Comment:
as the only user-facing API change, i'm wondering if we should be conservative here and call it something like `experimental_snapshot` or `preview_snapshot`. i'd like to think through the product implications of this feature a bit more before committing to it as public api in perpetuity.